uber/automated_emails_server.py

```diff
@@ -389,43 +388,45 @@ def get_pending_email_data():
     Returns: A dict of senders -> email idents -> pending counts for any email category with pending emails,
     or None if none are waiting to send or the email daemon service has not finished any runs yet.
     """
-    has_email_daemon_run_yet = SendAllAutomatedEmailsJob.last_result.get('completed', False)
-    if not has_email_daemon_run_yet:
-        return None
 
-    categories_results = SendAllAutomatedEmailsJob.last_result.get('categories', None)
-    if not categories_results:
-        return None
+    with Session() as session:
+        if not EmailDaemonStatus.last_result_looks_valid(session):
+            return None
+
+        categories_results = session.query(EmailDaemonCategoryResult).all()
+        if len(categories_results) <= 0:
+            return None
 
-    pending_emails_by_sender = defaultdict(dict)
+        pending_emails_by_sender = defaultdict(dict)
 
-    for automated_email in AutomatedEmail.instances.values():
-        sender = automated_email.sender
-        ident = automated_email.ident
+        for automated_email in AutomatedEmail.instances.values():
+            sender = automated_email.sender
+            ident = automated_email.ident
 
-        category_results = categories_results.get(ident, None)
-        if not category_results:
-            continue
+            category_results = next((result for result in categories_results if result.ident == ident), None)
+            if not category_results:
+                continue
 
-        unsent_because_unapproved_count = category_results.get('unsent_because_unapproved', 0)
-        if unsent_because_unapproved_count <= 0:
-            continue
+            if category_results.unsent_because_unapproved <= 0:
+                continue
+
+            pending_emails_by_sender[sender][ident] = {
+                'num_unsent': category_results.unsent_because_unapproved,
+                'subject': automated_email.subject,
+                'sender': automated_email.sender,
+            }
 
-        pending_emails_by_sender[sender][ident] = {
-            'num_unsent': unsent_because_unapproved_count,
-            'subject': automated_email.subject,
-            'sender': automated_email.sender,
-        }
 
     return pending_emails_by_sender
 
 
 uber.scheduler.register_task(
-    lambda: uber.scheduler.schedule.every(5).minutes.do(SendAllAutomatedEmailsJob.send_all_emails),
+    lambda: uber.scheduler.schedule.every(30).seconds.do(SendAllAutomatedEmailsJob.send_all_emails),
```

Comment:
since the scheduler (and the email daemon itself) won't let overlapping jobs happen, no reason to not just shorten this window.